experimental/overhead_matching/swag/data/vigor_dataset.py

```diff
@@ -98,6 +102,51 @@ def __getitem__(self, idx):
     def __len__(self):
         return len(self._panorama_metadata)
     
+    def generate_random_path(self,
+                             generator: torch.Generator,
+                             max_length_m: float,
+                             turn_temperature: float)->list[int]:
+        path = [] 
+        distance_traveled_m = 0.0
+        last_direction = torch.zeros(2)
+        # select start panorama
+        last_index = torch.randint(0, len(self), (1,), generator=generator)
+
+        def softmax_w_temp(x: torch.Tensor, t: float)-> torch.tensor:
+            ex = torch.exp(x / t)
+            return ex / torch.sum(ex)
+
+        while distance_traveled_m < max_length_m:
+            path.append(last_index)
+            last_item = self[last_index]
```

Comment:
Consider indexing into `self._panorama_metadata.loc[last_index]` instead of `self[last_index]`. using `__get_index__()`  ends up reading the image, which doesn't appear to be necessary.